### PR TITLE
osutil: fix parsing /proc/PID/mounts with spaces

### PR DESCRIPTION
**Pull request by zyga**
The 9p file system bugs strike again!

This is a variation of 3d72b032669cab29fdf639bc32894ac258825fa4 (osutil: fix parsing super-block options with spaces) but for /proc/PID/mounts.

The problem is somewhat worse here, as unlike /proc/self/mountinfo, the mount profile parser is used to load both /etc/fstab as well as /proc/{,PID/}mounts. In the former case it is technically allowed to omit the last several fields (options and two numbers) but in the latter case they are always emitted.

The problem is somewhat unsolvable, as the format never expected to handle any fields with unescaped spaces. We pick a pragmatic approach that is certainly valid but not the only way to parse a mount entry.

In other news, in 2024 Linux still cannot handle text formats without running into Johnny Tables issues.

Jira: https://warthogs.atlassian.net/browse/SNAPDENG-24318

### CODE REVIEW

File: osutil/mountentry_linux_test.go [hunk 1]
```diff
@@ -218,6 +216,30 @@ func (s *entrySuite) TestParseMountEntry6(c *C) {
...
+	c.Assert(err, IsNil)
+
+	c.Assert(e.Name, Equals, "C:\\Program Files\\Docker\\Docker\\resources")
+	c.Assert(e.Dir, Equals, "/Docker/host")
+	c.Assert(e.Type, Equals, "9p")
+	c.Assert(e.Options, DeepEquals, []string{
+		"rw",
+		"dirsync",
+		"noatime",
+		"aname=drvfs;path=C:\\Program Files\\Docker\\Docker\\resources;symlinkroot=/mnt/",
```
Comment (bboozzoo) [inline osutil/mountentry_linux_test.go:231]: it'd be nice to figure out where this is coming from and maybe fix the kernel side as well, looking at https://elixir.bootlin.com/linux/v6.10/source/fs/9p/v9fs.c#L116 the kernel bits may need some escaping

**Review by bboozzoo — APPROVED** [review]
LGTM

File: osutil/mountentry_linux_test.go (hunk 1 above)
Comment (zyga) [reply osutil/mountentry_linux_test.go:231]: I've prepared a kernel patch and I'll send it once I figure out the right mailing list.

**Review by alexmurray — APPROVED** [review]
LGTM - thanks for the details commit message and in-code comment regarding this - from my reading of `man fstab(5)` this looks correct and is a good pragmatic solution.

File: osutil/mountentry_linux.go [hunk 2]
```diff
@@ -40,29 +40,60 @@ func ParseMountEntry(s string) (MountEntry, error) {
...
+			// The problem is further complicated by the 2nd optional field, so
+			// imagine seeing 5 6 instead of just 5 in the example above.
+			//
+			// We try to be pragmatic and observe the following properties:
+			//
+			// - The /proc/PID/mounts file does not hide optional fields.
+			// - The /etc/fstab file doesn't usually use unescaped spaces.
+			//
+			// As such, the ambiguous cases with five or six fields are handled
+			// as if mount options had no spaces.
```
Comment (alfonsosanchezbeato) [inline osutil/mountentry_linux.go:75]: Is the comment right? What I see is that fields fs_freq and fs_passno are expected to have no spaces, which is correct.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, a nitpick about a comment though

File: osutil/mountentry_linux.go (hunk 2 above)
Comment (zyga) [reply osutil/mountentry_linux.go:75]: Yes, those two fields do not have spaces. The only problem is with options. Perhaps ther comment is unclear. Can you suggest an edit?

Comment (alfonsosanchezbeato) [reply osutil/mountentry_linux.go:75]:
```suggestion
			// As such, the ambiguous cases with five or six fields are handled
			// as if mount options (fs_mntops in fstab(5)) had no spaces.
```

Comment (zyga) [reply osutil/mountentry_linux.go]: Thanks, applied.

**Pull request merged**